cohort-engine/src/main/java/com/ibm/cohort/engine/measure/cache/TransientRetrieveCacheContext.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.engine.measure.cache;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+public class TransientRetrieveCacheContext implements RetrieveCacheContext {
+
+	private Cache<CacheKey, Iterable<Object>> currentCache;
+
+	@Override
+	public Cache<CacheKey, Iterable<Object>> newCache(String patientId) {
+		this.currentCache = Caffeine.newBuilder()
+				.recordStats()
+				.maximumSize(1_000)
```

Comment:
Any logic around 1000, or is  this just a starting point so that you can start experimenting?

Reply:
Purely a silly default to get me running.